games/pacman.puml

```diff
@@ -0,0 +1,153 @@
+@startuml
+package "Process Input Handler" {
+!includesub ../uml/frontend-software-system/process-input-handler.puml!INTERFACE
+}
+package "Render Handler" {
+!includesub ../uml/frontend-software-system/render-handler.puml!INTERFACE
+}
+!include ../uml/frontend-software-system/game-settings.puml
+
+
+package Pac-Man {
+
+class Position {
+ - x : int
+ - y : int
+}
+
+class Wall {
+ - sprite : Sprite
+ - position: Position
+}
+
+class PowerSphere
+{
+ - sprite : Sprite
+ - time : int
+ - position: Position
+ + collect() : int
+ + activatePower() : void
+}
+
+class Sphere
+{
+ - sprite : Sprite
+ - position: Position
+ + collect() : int
+}
+
+class Ghost
+{
+ - sprite : Sprite
+ -state : string
+ -color : string
+ + fleePacman () : void
+ + chasePacman () : void
+}
+
+interface ClydeGhost{}
+interface PinkyGhost{}
+interface InkyGhost{}
+interface BlinkyGhost{}
+
+interface ICharacter
+{
+ - position: Position
+ + respawn (Position position) : void
+ + move (Position) : bool
+}
+class Pacman 
+{
+ - sprite : Sprite
+ - lives : int
+ - direction : string
+ + changeDirection(string direction) : void
+ + looseLife() : bool
+}
+
+class Maze
```

Comment:
wouldn't it be better to represent it as a matrix
    it could be an aggregation of blocks
    where each block can be an ISphere or an ICharacter. And an aggregation of blocks would be a wall.